bears/generate_package.py

```diff
@@ -113,7 +117,13 @@ def main():
     else:
         bear_version = repr(bear_version) + '.' + str(int(time.time()))
 
-    for bear_file_name in sorted(set(glob('bears/**/*Bear.py'))):
+    if args.bear:
+        bear_files = [f for f in sorted(set(glob('bears/**/*Bear.py')))
+                      for x in args.bear if x in f]
```

Comment:
This is quite complicated to read, could you please add a comment explaining what's happening here.